client.go

```diff
@@ -86,6 +90,13 @@ func (c *singleClient) DoMulti(ctx context.Context, multi ...Completed) (resps [
 	attempts := 1
 retry:
 	resps = c.conn.DoMulti(ctx, multi...).s
+	if c.hasConnLftm {
+		for _, resp := range resps {
+			if resp.Error() == errConnExpired {
```

Comment:
Is it possible that `errConnExpired` happens in the middle of `DoMulti`? I am not sure, but If it is possible then we should not retry preceding requests that don't receive the error.

Reply:
Ah,  I think it's unlikely. Surely all responses have same error when changing `p.state`.

I will change that like the following.

```
if resps[0].Error() == errConnExpired {
  goto retry
}
```

Reply:
Fixed [c0c3657](https://github.com/redis/rueidis/pull/727/commits/c0c365740db9eeeae1d0833a475b1737078d8423)

Reply:
ok, could you leave a comment in the code to explain why it won't happen?

Reply:
When I was checking the behavior of connection lifetime on concurrent process and then I found the error `read tcp [::1]:35190->[::1]:6379: use of closed network connection`  through `singleClient.DoMulti`. I think probably the error occurred because of `pipe.Close()` , but the investigation isn't going well. Could you advice me for that? @rueian

Reply:
Hi @terut, I'm sorry that I just saw your message. I think the problem could be in the `_backgroundRead`. It returns the network error directly without checking the p.Error().

Reply:
https://github.com/redis/rueidis/blob/b799437d0fd865cf2c1342a4edbf82abecc66d77/pipe.go#L490
https://github.com/redis/rueidis/blob/b799437d0fd865cf2c1342a4edbf82abecc66d77/pipe.go#L616

Reply:
@rueian Thanks! As you said, it looks like that `_backgroundRead` returns that error.

I will change that lines to return `errConnExpired` when expired and then `errConnExpired` happens in the middle of `DoMulti`, so we may should check the error of all response.

Reply:
Hi @terut, any update?

Reply:
Sorry for late. I had no time to spare... I think I can work on this problem from this week. @rueian

Reply:
> Sorry for late. I had no time to spare... I think I can work on this problem from this week.

No worries. 

> Sometime incremented value is 10001, 10002 and so on when loop is 10000.

What do you mean by this? I think this can be a general feature for those who want a limited lifetime on each connection for any reason.

Reply:
When I just counted up for 10000 times using connection lifetime option, the value of keys is over 10000. But my implementation is not correct for now at the point of view of the error handling of `_backgroundRead`, I will try to count up again after implementing correctly.